mozilla/fx-private-relay-add-on
# [MPP-1714] Popup Refactor - Report Issue Panel

Blocked by: https://github.com/mozilla/fx-private-relay-add-on/pull/455

## Summary

Add "Report Issue" panel to add-on toolbar

Links: 
- Jira Epic: https://mozilla-hub.atlassian.net/browse/MPP-1708
- [Figma designs](https://www.figma.com/file/NP5tLeIFJeHC5HvaqpLQdp/MPPUX-129-Add-On?node-id=144%3A5071&t=MN61ZFZ9w2429nNj-4) 

## Tests

1. Submit report while on valid page
- Run add-on
- Sign-in
- Go to https:mozilla.org
- Open popup > Gear icon > Report Issue
- **Expected:** Panel should be open to report issue, with mozilla.org prefilled.
- Submit form
- **Expected:** Panel should change to success field
- Click "Continue"
- **Expected:** User should be back on Settings panel 

1. Submit report while on invalid page
- Run add-on
- Sign-in
- Go to internal page: `about:about` in Firefox (or `chrome://about/` in Chrome) 
- Open popup > Gear icon > Report Issue
- **Expected:** Panel should be open to report issue, with nothing prefilled. The submit button should be disabled
- Add web address to URL input field
- **Expected:** Submit button should remove disabled state
- Submit form
- **Expected:** Panel should change to success field
- Click "Continue"
- **Expected:** User should be back on Settings panel 

## Screenshots

Valid report (before submit):
<img width="390" alt="image" src="https://user-images.githubusercontent.com/2692333/214168568-80af7370-46c8-4f07-b08d-365d2db6f34e.png">

Invalid report (on internal/non-HTTP page):
<img width="387" alt="image" src="https://user-images.githubusercontent.com/2692333/214168820-6bfac690-be1f-415f-97c4-323269c3a228.png">

Completed report: 
<img width="395" alt="image" src="https://user-images.githubusercontent.com/2692333/214168885-a88b89ee-2452-4f6a-b19e-2a88cedaa10a.png">
